File: tinyflix-back/src/models/movie.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct Movie {
    pub id: String,
    pub title: String,
    pub genre: String,
    pub director: String,
    pub actors: Vec<String>,
    pub rating: Option<f32>,
    pub s3_path: String,
}

File: tinyflix-back/models/movie.py
class Movie:
    def __init__(self, id, title, genre, director, actors, rating, s3_path):
        self.id = id
        self.title = title
        self.genre = genre
        self.director = director
        self.actors = actors
        self.rating = rating
        self.s3_path = s3_path


    def to_dict(self):
        return {
            'id': self.id,
            'title': self.title,
            'genre': self.genre,
            'director': self.director,
            'actors': self.actors,
            'rating': self.rating,
            's3_key': self.s3_key
        }

